tests/timeseries/test_periodic_feature_extractor.py

```diff
@@ -0,0 +1,140 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for the timeseries averager."""
+
+from datetime import datetime, timedelta, timezone
+from typing import List
+
+import numpy as np
+from frequenz.channels import Broadcast
+
+from frequenz.sdk.timeseries import MovingWindow, PeriodicFeatureExtractor, Sample
+from frequenz.sdk.timeseries._base_types import UNIX_EPOCH
+from tests.timeseries.test_moving_window import init_moving_window, push_lm_data
+
+
+async def init_feature_extractor(
+    data: List[float], period: timedelta
+) -> PeriodicFeatureExtractor:
+    """
+    Initialize a PeriodicFeatureExtractor with a `MovingWindow` that contains the data.
+
+    Args:
+        data: The data that is pushed into the moving window.
+        period: The distance between two successive windows.
+
+    Returns:
+        PeriodicFeatureExtractor
+    """
+    window, sender = init_moving_window(timedelta(seconds=len(data)))
+    await push_lm_data(sender, data)
+
+    return PeriodicFeatureExtractor(moving_window=window, period=period)
+
+
+async def init_feature_extractor_no_data(period: int) -> PeriodicFeatureExtractor:
+    """
+    Initialize a PeriodicFeatureExtractor with a `MovingWindow` that contains no data.
+
+    Args:
+        period: The distance between two successive windows.
+
+    Returns:
+        PeriodicFeatureExtractor
+    """
+    # We only need the moving window to initialize the PeriodicFeatureExtractor class.
+    lm_chan = Broadcast[Sample]("lm_net_power")
+    # monkey patch the MovingWindow class to have a fixed size such that
+    # it can be initialized with a distance bigger then the window size.
+    MovingWindow.__len__ = lambda self: period  # type: ignore[method-assign]
```

Comment:
You should use `with mock.patch(...)` or the `mocker` *fixture* so the `MovingWindow.__len__` is restored after the test, otherwise you can run to really weird shit.

Reply:
Actually in that case the patching is not needed any longer, since the semantics of period have changed.